integrations/pinterest-tag/lib/index.js

```diff
@@ -1,68 +1,80 @@
-'use strict'
+'use strict';
 
 /**
  * Module dependencies.
  */
 
-var integration = require('@segment/analytics.js-integration')
-var analyticsEvents = require('analytics-events')
+var integration = require('@segment/analytics.js-integration');
+var analyticsEvents = require('analytics-events');
 
 /**
  * Expose `Pinterest` integration.
  */
 
-var Pinterest = module.exports = integration('Pinterest Tag')
+var Pinterest = (module.exports = integration('Pinterest Tag')
   .global('pintrk')
   .option('tid', '')
   .option('pinterestCustomProperties', [])
   .mapping('pinterestEventMapping')
-  .tag('<script src="https://s.pinimg.com/ct/core.js"></script>')
+  .tag('<script src="https://s.pinimg.com/ct/core.js"></script>'));
 
-Pinterest.prototype.initialize = function () {
+Pinterest.prototype.initialize = function() {
   // We require a Tag ID to run this integration.
-  if (!this.options.tid) return
+  if (!this.options.tid) return;
 
   // Preparation for loading the Pinterest script.
   (function(e){if(!window.pintrk){window.pintrk=function(){window.pintrk.queue.push(Array.prototype.slice.call(arguments))};var n=window.pintrk;n.queue=[],n.version="3.0";}})(); // eslint-disable-line
 
-  this.load(this.ready)
-  pintrk('load', this.options.tid)
-  pintrk('page') // This is treated semantically different than our own page implementation.
+  this.load(this.ready);
+  window.pintrk('load', this.options.tid);
+  window.pintrk('page'); // This is treated semantically different than our own page implementation.
 
-  this.createPropertyMapping()
-}
+  this.createPropertyMapping();
+};
 
-Pinterest.prototype.loaded = function () {
-  return !!(window.pintrk && window.pintrk.queue && window.pintrk.queue.push !== Array.prototype.push)
```

Comment:
This was actually never resolving to `true`. The pinterest tag seems to load another tag so there's no way we can actually hook into that script load lifecycle and delay initializing until it's finished.

That being said it's not necessary to because we interface with their synchronous API (`pintrk()`) when triggering events so we don't need to wait until their full script loads to start tracking data.